### PR TITLE
[SCB-891] Add kamon metrics to Omega client

### PR DESCRIPTION
**Pull request by maheshrajus**
Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/SCB) filed for the change (usually before you start working on it).  Trivial changes like typos do not require a JIRA issue.  Your pull request should address just this issue, without pulling in other changes.
 - [x] Each commit in the pull request should have a meaningful subject line and body.
 - [x] Format the pull request title like `[SCB-XXX] Fixes bug in ApproximateQuantiles`, where you replace `SCB-XXX` with the appropriate JIRA issue.
 - [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [x] Run `mvn clean install` to make sure basic checks pass. A more thorough check will be performed on your pull request automatically.
 - [x] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---

### CODE REVIEW

**Comment by WillemJiang** [comment]
We need to find a way to build trace stack to help us find the bottleneck not just enable kamon.

**Comment by zhfeng** [comment]
@WillemJiang can you add your review as the changes request ? Btw, do you have any suggest to build this trace stack ?
It might be useful to add some references to help @maheshrajus  to understand.

**Comment by WillemJiang** [comment]
@zhfeng  I agree with you, we need to provide more specific information about the code.
But current solution of the kamon metrics doesn't meet the need of finding the bottleneck, we may need to find another solution for it.  That is why didn't merge the code.

**Comment by maheshrajus** [comment]
Kamon will help us to find the metrics related to major flows [omega and alpha events, compensation notifications].

File: omega/omega-transport/omega-transport-dubbo/src/main/java/org/apache/servicecomb/saga/omega/transport/dubbo/SagaDubboConsumerFilter.java [hunk 1]
```diff
@@ -36,10 +36,14 @@
...
 import static org.apache.servicecomb.saga.omega.context.OmegaContext.LOCAL_TX_ID_KEY;
 
+import kamon.annotation.EnableKamon;
+import kamon.annotation.Trace;
+
 /**
  * add saga transaction id to dubbo invocation
  */
 @Activate(group = {Constants.CONSUMER})
+@EnableKamon
```
[the remark this answers is not in the document]
Comment (maheshrajus) [reply omega/omega-transport/omega-transport-dubbo/src/main/java/org/apache/servicecomb/saga/omega/transport/dubbo/SagaDubboConsumerFilter.java]: ok, I will remove for all transport flows

File: saga-demo/saga-spring-demo/pom.xml [hunk 2]
```diff
@@ -43,5 +43,39 @@
...
+    <dependency>
+      <groupId>io.kamon</groupId>
+      <artifactId>kamon-core_2.12</artifactId>
+    </dependency>
+    <dependency>
+      <groupId>io.kamon</groupId>
+      <artifactId>kamon-annotation_2.12</artifactId>
+    </dependency>
+    <dependency>
+      <groupId>com.typesafe.akka</groupId>
```
[the remark this answers is not in the document]
Comment (maheshrajus) [reply saga-demo/saga-spring-demo/pom.xml]: I tested it, without this dependency it is not working

**Comment by maheshrajus** [comment]
As per discussion with willeam jiang, we don't think kamon can provide the monitor service that we want.  We need to check some other frame work for this. So closing this PR

**Pull request closed without merging**